sdks/android/Makefile

```diff
@@ -79,3 +63,146 @@ logcat:
+
+$(APK_DIR) $(APK_DIR)/jni $(APK_DIR)/jni/armeabi-v7a $(APK_DIR)/jni/arm64-v8a $(APK_DIR)/jni/x86 $(APK_DIR)/jni/x86_64 $(APK_DIR)/assets $(APK_DIR)/assets/asm:
+	mkdir -p $@
+
+$(APK_DIR)/AndroidManifest.xml: $(APK_DIR)/%: % | $(APK_DIR)
+	cp $< $@
+
+$(APK_DIR)/build.xml $(APK_DIR)/local.properties $(APK_DIR)/project.properties: $(APK_DIR)/AndroidManifest.xml | $(APK_DIR)
+	cd $(APK_DIR) && $(ANDROID) update project -p . -t "android-14"
+
+.PHONY: res/ src/
+$(APK_DIR)/res/ $(APK_DIR)/src/: $(APK_DIR)/%: % | $(APK_DIR)
+	cp -r $< $@
+
+$(APK_DIR)/jni/Android.mk $(APK_DIR)/jni/Application.mk $(APK_DIR)/jni/runtime-bootstrap.c: $(APK_DIR)/%: % | $(APK_DIR)/jni
+	cp $< $@
+
+MONO_LIBS= \
+	libmonosgen-2.0.so \
+	libMonoPosixHelper.so
+
+$(patsubst %,$(APK_DIR)/jni/armeabi-v7a/%,$(MONO_LIBS)): $(APK_DIR)/jni/armeabi-v7a/%: $(TOP)/sdks/out/android-arm/lib/% | $(APK_DIR)/jni/armeabi-v7a
+	cp $< $@
+
+$(patsubst %,$(APK_DIR)/jni/arm64-v8a/%,$(MONO_LIBS)): $(APK_DIR)/jni/arm64-v8a/%: $(TOP)/sdks/out/android-arm64/lib/% | $(APK_DIR)/jni/arm64-v8a
+	cp $< $@
+
+$(patsubst %,$(APK_DIR)/jni/x86/%,$(MONO_LIBS)): $(APK_DIR)/jni/x86/%: $(TOP)/sdks/out/android-x86/lib/% | $(APK_DIR)/jni/x86
+	cp $< $@
+
+$(patsubst %,$(APK_DIR)/jni/x86_64/%,$(MONO_LIBS)): $(APK_DIR)/jni/x86_64/%: $(TOP)/sdks/out/android-x86_64/lib/% | $(APK_DIR)/jni/x86_64
+	cp $< $@
+
+$(APK_DIR)/assets/mconfig: machine.config | $(APK_DIR)/assets
+	cp $< $@
+
+$(APK_DIR)/assets/asm/main.exe: managed/main.cs | $(APK_DIR)/assets/asm
+	mcs /nostdlib -unsafe -out:$@ $(BCL_REFERENCES) $(NUNIT_REFERENCES) $<
+
+$(APK_DIR)/assets/asm/Mono.Android.dll: managed/fake-monodroid.cs | $(APK_DIR)/assets/asm
+	mcs /nostdlib -target:library -out:$@ $(BCL_REFERENCES) $<
+
+$(patsubst %,$(APK_DIR)/assets/asm/%,$(BCL_LIBS)) $(patsubst %,$(APK_DIR)/assets/asm/%,$(NUNIT_LIBS)): $(APK_DIR)/assets/asm/%: $(TOP)/sdks/out/bcl/monodroid/% | $(APK_DIR)/assets/asm
+	cp $< $@
+
+.PHONY: package
+package: $(APK_DIR)/AndroidManifest.xml
+package: $(APK_DIR)/build.xml
+package: $(APK_DIR)/local.properties
+package: $(APK_DIR)/project.properties
+package: $(APK_DIR)/res/
+package: $(APK_DIR)/src/
+package: $(APK_DIR)/jni/Android.mk
+package: $(APK_DIR)/jni/Application.mk
+package: $(APK_DIR)/jni/runtime-bootstrap.c
+package: $(patsubst %,$(APK_DIR)/jni/armeabi-v7a/%,$(MONO_LIBS))
+package: $(patsubst %,$(APK_DIR)/jni/arm64-v8a/%,$(MONO_LIBS))
+package: $(patsubst %,$(APK_DIR)/jni/x86/%,$(MONO_LIBS))
+package: $(patsubst %,$(APK_DIR)/jni/x86_64/%,$(MONO_LIBS))
+package: $(APK_DIR)/assets/mconfig
+package: $(APK_DIR)/assets/asm/main.exe
+package: $(APK_DIR)/assets/asm/Mono.Android.dll
+package: $(patsubst %,$(APK_DIR)/assets/asm/%,$(BCL_LIBS))
+package: $(patsubst %,$(APK_DIR)/assets/asm/%,$(NUNIT_LIBS))
+package: $(patsubst %,$(APK_DIR)/assets/asm/monodroid_%_test.dll,$(CHECKS))
+
+.PHONY: build-%
+build-%: package
+	printf "monodroid_$*_test.dll" > $(APK_DIR)/assets/asm/testassembly.txt
```

Comment:
This allows only a single test assembly per app. Why is perfect for local testing. OTOH, which XTC testing, having a single app makes it easy to manage execution and results collection.

Reply:
I will bring the XTC capability back.